packages/wifi_basic/android/src/main/kotlin/dev/flutternetwork/wifi/wifi_basic/WifiBasicPlugin.kt

```diff
@@ -0,0 +1,81 @@
+package dev.flutternetwork.wifi.wifi_basic
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.wifi.WifiManager
+import android.os.Build
+import android.provider.Settings
+import androidx.annotation.NonNull
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+/** WifiBasicPlugin */
+class WifiBasicPlugin : FlutterPlugin, MethodCallHandler {
+    /// The MethodChannel that will the communication between Flutter and native Android
+    ///
+    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+    /// when the Flutter Engine is detached from the Activity
+    private lateinit var channel: MethodChannel
+    private var context: Context? = null
```

Comment:
Should be `lateinit` and non-nullable.

Reply:
Have kept it nullable so this it can be referenced null back in detach.

Reply:
To avoid any memory leak.

Reply:
Application Contexts do not leak memory. They are global for the entire app process. You're already getting an app context from `FlutterPlugin.FlutterPluginBinding`. If anything, maybe the `WifiManager` can be nullable.

Reply:
made suggested changes with 
 https://github.com/alternadom/WiFiFlutter/pull/201/commits/a38aef30db9a81dbf0f9a7f8eb9001475af54c3a